mtv-plan/mtv.sw.yaml

```diff
@@ -0,0 +1,93 @@
+specVersion: "0.8"
+id: mtv-plan
+version: '1.0'
+name: MTV plan assessment workflow
+description: assessment creates a plan and waits for it to be ready or failed
+annotations:
+  - "workflow-type/assessment"
+dataInputSchema: schemas/mtv-input.json
+start: CreatePlan
+functions:
+  - name: createPlan
+    type: rest
+    operation: 'specs/mtv.yaml#createPlan'
+  - name: getPlan
+    type: rest
+    operation: 'specs/mtv.yaml#getPlan'
+  - name: getPlanError
+    type: expression
+    operation: .getStatusResult.status.conditions | map(select(.category == "Error" or .category == "Critical"))[0].message
+states:
+  - name: CreatePlan
+    type: operation
+    actions:
+      - functionRef:
+          refName: createPlan
+          arguments:
+            namespace: .migrationNamespace
+            apiVersion: "forklift.konveyor.io/v1beta1"
+            kind: Plan
+            metadata:
+              name: .migrationName
+              namespace: .migrationNamespace
+            spec:
+              map:
+                network:
+                  name: .networkMap
+                  namespace: .migrationNamespace
+                storage:
+                  name: .storageMap
+                  namespace: .migrationNamespace
+              provider:
+                destination:
+                  name: .destinationProvider
+                  namespace: .migrationNamespace
+                source:
+                  name: .sourceProvider
+                  namespace: .migrationNamespace
+              targetNamespace: .migrationNamespace
+              vms: .vms
+    transition: GetPlan
+  - name: GetPlan
+    type: operation
+    actions:
+      - name: getPlan
+        actionDataFilter:
+          toStateData: .getStatusResult
+        functionRef:
+          refName: getPlan
+          arguments:
+            namespace: .migrationNamespace
+            plan: .migrationName
+        sleep:
+          before: PT10S
+    transition: CheckPlanStatus
+  - name: CheckPlanStatus
```

Comment:
Would it be interesting to have a global timeout or at least a maximum amount of retries? ForM2K for instance, I know that the plan can go wrong and block the workflow for ever, having some failsafe would be nice

Reply:
i was instructed by @pkliczewski to wait indefinitely and rely on MTV to reach a final state. https://redhat-internal.slack.com/archives/C059YMMB664/p1717945993380539?thread_ts=1717945371.811889&cid=C059YMMB664

Reply:
If we are confident that MTV will timeout, then it's fine

We could use global timeout to be extra sure that after some time, the workflow will go into error and free its resources

Reply:
SonataFlow already has that

Reply:
Added a note in the first section of the README file about us relying on MTV